src/psyclone/tests/psyir/transformations/read_only_verify_test.py

```diff
@@ -0,0 +1,129 @@
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author: J. Henrichs, Bureau of Meteorology
+# -----------------------------------------------------------------------------
+
+''' Module containing tests for ReadOnlyVerifyTrans and ReadOnlyVerifyNode
+'''
+
+from __future__ import absolute_import
+
+import pytest
+
+from psyclone.errors import InternalError
+from psyclone.psyir.nodes import Node, ReadOnlyVerifyNode
+from psyclone.psyir.transformations import (ReadOnlyVerifyTrans,
+                                            TransformationError)
+from psyclone.tests.utilities import get_invoke
+from psyclone.transformations import OMPParallelLoopTrans
+
+# --------------------------------------------------------------------------- #
+# ================== Extract Transformation tests =========================== #
+# --------------------------------------------------------------------------- #
+
+
+def test_extract_trans():
+    '''Tests basic functions in ReadOnlyVerifyTrans.'''
+    read_only = ReadOnlyVerifyTrans()
+    assert str(read_only) == "Create a sub-tree of the PSyIR that has " \
+                             "a ReadOnlyVerifyNode at its root."
+    assert read_only.name == "ReadOnlyVerifyTrans"
+
+
+# -----------------------------------------------------------------------------
+def test_malformed_extract_node(monkeypatch):
+    ''' Check that we raise the expected error if an ReadOnlyVerifyNode does
+    not have a single Schedule node as its child. '''
+    read_node = ReadOnlyVerifyNode()
+    monkeypatch.setattr(read_node, "_children", [])
+    with pytest.raises(InternalError) as err:
+        _ = read_node.read_only_verify_body
+    assert "malformed or incomplete. It should have a " in str(err.value)
+    monkeypatch.setattr(read_node, "_children", [Node(), Node()])
+    with pytest.raises(InternalError) as err:
+        _ = read_node.read_only_verify_body
+    assert "malformed or incomplete. It should have a " in str(err.value)
+
+
+# -----------------------------------------------------------------------------
+def test_read_only_basic():
+    '''Check basic functionality: node names, schedule view.
+    '''
+    _, invoke = get_invoke("test11_different_iterates_over_one_invoke.f90",
+                           "gocean1.0", idx=0)
+    read_only = ReadOnlyVerifyTrans()
+    new_sched, _ = read_only.apply(invoke.schedule[0].loop_body[0])
+    new_sched.view()
```

Comment:
Probably worth using `capsys` to check that the result of the `view` call at least contains some of the things we expect.

Reply:
I think the ``view()`` was more for debugging (or from the original test I used), but it can't hurt to add more tests, so all done.

Reply:
Nice, thanks.